### PR TITLE
Wip 19441 java optional in javadsl drewhk

### PR DESCRIPTION
**Pull request by drewhk**

### CODE REVIEW

**Comment by drewhk** [comment]
Migration guide is coming next

**Comment by drewhk** [comment]
what? This compiled locally what is the isse now...

**Comment by patriknw** [comment]
Here is another one: https://github.com/akka/akka/blob/master/akka-stream/src/main/scala/akka/stream/Attributes.scala#L69

**Comment by drewhk** [comment]
Hm, that was not a japi.Option hence I missed it. Thanks for noticing, this was broken even in 2.0.x (but fixing this would not be bin compat)

**Comment by patriknw** [comment]
LGTM

**Comment by drewhk** [comment]
PLS BUILD

**Comment by drewhk** [comment]
Validation is down?

**Comment by ktoso** [comment]
hm let's restart jenkins then, willdo

**Comment by drewhk** [comment]
I don't think the test failure is related.

**Comment by rkuhn** [comment]
LGTM (just skimmed)

File: akka-stream/src/main/scala/akka/stream/impl/Emit.scala [hunk 1]
```diff
@@ -1,33 +0,0 @@
...
- * Copyright (C) 2014 Typesafe Inc. <http://www.typesafe.com>
- */
-package akka.stream.impl
-
-import scala.collection.immutable
-
-/**
- * INTERNAL API
- */
-private[akka] trait Emit { this: ActorProcessorImpl with Pump ⇒
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/Emit.scala]: was unused?

**Comment by ktoso** [comment]
LGTM, fix for unrelated failure on it's way in already: https://github.com/akka/akka/issues/19464

**Comment by ktoso** [comment]
Refs #19441

File: akka-stream/src/main/scala/akka/stream/impl/Emit.scala (hunk 1 above)
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/Emit.scala]: this one yes, the others are sadly used by groupBy and TLS.

**Pull request merged**